src/entity/Cart.ts

```diff
@@ -0,0 +1,21 @@
+import {Entity, PrimaryGeneratedColumn, Column, ManyToOne, ManyToMany, JoinTable, OneToMany} from "typeorm"
+import {Users} from "./Users";
+import {Products} from "./Products";
+import {CartItem} from "./CartItem";
+
+@Entity()
```

Comment:
There's auxiliary types in typeORM like DeletedAt, CreatedAt, UpdatedAt. When using softDelete, column DeletedAt will be automatically updated with deleted data. I highly recommend to use it.